### PR TITLE
DS3/DS4/Dualsense: Improvements

### LINKED ISSUE #15519: (Windows) DualSense LED goes black after stopping a game
### Quick summary

In Windows, after stopping a game in RPCS3, the LED on the connected DualSense goes black--regardless of the LED color setting under Configuration - Pad--making it hard to tell that the DualSense is still on. This persists after closing RPCS3 itself.

### Details

1) Have a DualSense controller on and wirelessly connected in Windows
2) Run RPCS3
3) If you haven't already got the DualSense set up in RPCS3, set "Handlers = DualSense" under "Configuration - Pads"
4) If desired, in that same window set "LED Settings - LED Color" to a color other than the default black
5) Start a PS3 game in RPCS3 -- the LEDs on the sides of the DualSense touchpad change to the color set in "Configuration - Pads - LED Settings"
6) Press the Stop button on RPCS3's master window to stop the game
== The DualSense LEDs go black (regardless of the "LED Color" setting)
7) Close RPCS3
== The DualSense LEDs stay black

At this point, it is very hard to tell that the DualSense is still on, making this something of an inconvenience.

### Attach a log file

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/15140559/RPCS3.log.gz)

### Attach capture files for visual issues

_No response_

### System configuration

Operating System:
Windows 11

CPU:
i7-10750H

GPU:
GTX 1660 Ti
Driver version 535.98

Other:
16 GB RAM
S271HL NVIDIA High Definition Audio

### Other details

_No response_

### PR DESCRIPTION
**Pull request by Megamouse**
- Dualsense: Don't turn off the lights when the game closes.
- Replace raw buffer access with struct casts
- Add more enum values
- Allow to access touchpads internally (not in use by rpcs3)

- Fixes #15519

### CODE REVIEW

File: rpcs3/Input/ds4_pad_handler.cpp [hunk 1]
```diff
@@ -235,26 +222,26 @@ std::unordered_map<u64, u16> ds4_pad_handler::get_button_values(const std::share
 	if (!ds4_dev)
 		return keyBuffer;
 
-	auto buf = ds4_dev->padData;
+	const ds4_input_report_common& input = *reinterpret_cast<const ds4_input_report_common*>(ds4_dev->padData.data());
```
Comment (elad335) [inline rpcs3/Input/ds4_pad_handler.cpp:225]: This is undefined behavior, use std::memcpy, read_from_ptr, write_to_ptr and std::bit_cast only when dealing with these.

Comment (elad335) [reply rpcs3/Input/ds4_pad_handler.cpp:225]: Such as 

```
ds4_input_report_common temp{};
temp.. do stuff;
write_to_ptr<ds4_input_report_common>(ds4_dev->padData.data(), &temp);

Comment (Megamouse) [reply rpcs3/Input/ds4_pad_handler.cpp]: I dont't really see the point in doing that for 1 byte aligned structs.
But I rewrote it to use a union for the input reports anyway.

Comment (elad335) [reply rpcs3/Input/ds4_pad_handler.cpp]: union is still ub though

Comment (Megamouse) [reply rpcs3/Input/ds4_pad_handler.cpp]: I don't see how anything in the current code is undefined

Comment (elad335) [reply rpcs3/Input/ds4_pad_handler.cpp]: Its ub, accessing more than one object through the same memory location. In c++, union is not about seamless casting, it provides lifetime for the first object written to it.
Its called "strict type aliasing rule"

Comment (Megamouse) [reply rpcs3/Input/ds4_pad_handler.cpp]: Should be fixed now

**Comment by qurious-pixel** [comment]
Dualsense pad handler is broken with https://github.com/RPCS3/rpcs3/pull/15522/commits/42ad241a5c2c0b908bdb2b303b9eb98d2d4d60ca but the LEDs stay on when closing RPCS3 when using Bluetooth. 
Wired is working for Dualsense and DS4. 
SDL handler works, and EvDev mostly works, except for motion and analog sticks.

**Comment by Megamouse** [comment]
Dualsense should be fixed now

**Comment by qurious-pixel** [comment]
Tested on rpcs3-v0.0.32-16595-af3af6d4 and Dualsense pad handler has been fixed for wired and bluetooth wireless.

**Pull request merged**